### PR TITLE
Add error handling in newHashKey

### PR DESCRIPTION
**Pull request by Mdaiki0730**
## Description
When calling the function newHashKey() if an error occurs while filling the new key with random data from the defined random source, the created hash key will be zeroed, which means, the content of the slice will be all 0's.

I think it's better to handle errors properly.

### CODE REVIEW

**Comment by torao** [comment]
Separating certain functions into a package for the purpose of passing coverage seems to me to be out of line with module design and doesn't seem to be a good idea. And I'm also concerned about the project-local implicit rules that arise.

* What was the key point of the problems
* What options were available
* What were the considerations
* Then, why did you decide this was the best approach?

File: libs/std/crypto/rand/rand.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package rand
```
Comment (torao) [inline libs/std/crypto/rand/rand.go:1]: No one looking at the source would know why this function is separated in this package. Any purpose or intent that can only be recognized outside of the source code should be noted in the comment.

**Review by torao — CHANGES_REQUESTED** [review]

**Comment by Mdaiki0730** [comment]
After thinking about the following, I am implementing it now.
- Problem: Error handling of crand.Read in newHashKey provides insufficient coverage, but it is difficult to write tests for it. 
- Options:
  - Ignore coverage → The convention of merging after all green is broken
  - Wrap crand.Read as a private function → A wrapper just for testing the standard package is born
  - Wrap crand.Read as public function in libs → Allows aggregation of packages that wrap functions for which test is essentially unnecessary


The current option is selected because the layer where the error occurs is not at the application level but is connected to the test ignore.
I would like to hear your opinion if there are better options for this.

**Comment by torao** [comment]
@Mdaiki0730 
Looking for other code that uses `crand.Read()`, There is a function `CRandBytes()` in `crypto/random.go` that could be used for this purpose. It may work the same as the current `newHashKey()`, so how about using this instead of creating a specific package?

https://github.com/line/ostracon/blob/main/crypto/random.go#L9-L22

**Comment by Mdaiki0730** [comment]
I think very good@@
Thank you for finding this function!

**Comment by Mdaiki0730** [comment]
I created another PR because we found a completely different solution than the this PR.
New One: https://github.com/line/ostracon/pull/509

**Pull request closed without merging**